src/views/CoCurricularTranscript/index.js

```diff
@@ -170,36 +186,33 @@ export default class Transcript extends Component {
 
     // sort activities by ActivityCode
     while (activities.length > 0) {
-      // shift removes and returns the first item in the array
-      let curAct = activities.shift();
-      let sessions = [curAct.SessionCode];
-
-      // Get the session codes for all activities matching the current activity's code,
-      // knowing activities will be sorted by session and Activity Code.
-      // Continue looking at activities while the activity at array front has the
-      // same activityCode as the current activity
-      while (activities.length > 0 && activities[0].ActivityCode === curAct.ActivityCode) {
-        sessions.push(activities.shift().SessionCode);
-      }
+      let curAct = activities[0];
 
-      // Translate the sessions array into a duration:
+      // keep track of the activity code which will be used to identify all activities of the same
+      // code so they can be grouped into one activity component
+      let curActCode = curAct.ActivityCode;
 
-      // Pop first session code from array and split into months and years
-      let sess = sessions.shift();
+      // Pop first session code from array and split into months and years, which are saved as
+      // the initial start and end dates
+      // (shift is like pop, removes and returns the first item in the array)
+      let sess = activities.shift().SessionCode;
       let startMon = this.sliceStart(sess);
       let endMon = this.sliceEnd(sess);
       let startYear = this.sliceYear(sess),
         endYear = startYear;
 
-      // look through the rest of the array, keeping only the start of the earliest session
-      // and the end of the latest session for streaks of consecutive sessions
+      // For each other activity matching curActCode, if it is consecutive to the current end date,
+      // save its end date as the new end date, otherwise, add the current start and end dates to
+      // the string 'duration' (because the streak is broken) and prepare to start a new streak.
+      // Loop assumes activities will be sorted by session and Activity Code.
       let duration = '';
-      let prevSess = sess;
-      while (sessions > 0) {
-        sess = sessions.shift();
-        if (false && prevSess) {
+      while (activities.length > 0 && activities[0].ActivityCode === curActCode) {
```

Comment:
The first half of this while loop is redundant with the while loop it is within. Maybe the above can be changed to an if, and these can be un-nested?